### PR TITLE
Auto remove/purge old repos and binaries

### LINKED ISSUE #109: automatic (configurable) time-to-deletion
_From @alfredodeza on May 24, 2016 19:3_

Binaries are huge, and for testing purposes there is no need to keep every binary and every repository built.

A configurable value (that defaults to two weeks) should be allowed to destroy a sha1/refs for a given project.

_Copied from original issue: ceph/mita#75_

### PR DESCRIPTION
**Pull request by alfredodeza**
Fixes #109

### CODE REVIEW

File: chacra/async/recurring.py [hunk 1]
```diff
@@ -53,6 +54,33 @@ def poll_repos():
...
+        return
+
+    logger.info('polling repos for purging....')
+    now = _now or datetime.datetime.utcnow()
+
+    # XXX Magical datetime, not user configurable
+    lifespan = now - datetime.timedelta(weeks=2)
+
+    for r in models.Repo.query.filter(models.Repo.modified < lifespan).all():
+        print r.modified
```
Comment (andrewschoen) [inline chacra/async/recurring.py]: I doubt you wanted to keep this print statement in here.

File: chacra/async/recurring.py [hunk 2]
```diff
@@ -53,6 +54,33 @@ def poll_repos():
...
+    """
+    if getattr(pecan.conf, 'purge_repos', False) is False:
+        logger.info('purge_repos option is unset or explicitly disabled, will skip purge')
+        return
+
+    logger.info('polling repos for purging....')
+    now = _now or datetime.datetime.utcnow()
+
+    # XXX Magical datetime, not user configurable
+    lifespan = now - datetime.timedelta(weeks=2)
```
Comment (andrewschoen) [inline chacra/async/recurring.py]: I know this is arbitrary, but are we sure 2 weeks is long enough?

**Comment by andrewschoen** [comment]
Looks good, but don't we need some ansible work here so we can deploy ``purge_repos=True`` to the test chacra instances?

Also, I'm curious how we're going to invalidate these repos in shaman once they've been deleted from chacra.

File: chacra/async/recurring.py (hunk 2 above)
Comment (alfredodeza) [inline chacra/async/recurring.py]: that is what I was told initially by @athanatos 

We can modify this easily if we need to I think

**Pull request merged**